contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -1641,3 +1644,55 @@ truncate_tsql_identifier(char *ident)
 					  PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
 
 }
+
+void
+rewrite_call_stmt(CallStmt *callstmt)
+{
+	PLExecStateCallStack *top_es_entry = NULL;
+	List*                name;
+	char                 *dbname = NULL;
+
+	if (exec_state_call_stack)
+		top_es_entry = exec_state_call_stack->next;
+
+	while (top_es_entry != NULL)
+	{
+		if (top_es_entry->estate && top_es_entry->estate->err_stmt &&
+			top_es_entry->estate->err_stmt->cmd_type == PLTSQL_STMT_EXEC &&
+			top_es_entry->estate->db_name != NULL)
+		{
+			dbname = pstrdup(top_es_entry->estate->db_name);
+			break;
+		}
+		top_es_entry = top_es_entry->next;
+	}
+
+	if (!dbname)
+		return;
+
+	name = callstmt->funccall->funcname;
+
+	switch (list_length(name))
+	{
+		case 2:
+			{
+				Node	   *schema = (Node *) linitial(name);
+
+				if (is_shared_schema(strVal(schema)))
+					break;		/* do nothing for shared schemas */
+
+				name = lcons(makeString(dbname), name);
+				break;
+			}
+		case 1:
+			{
+				name = lcons(makeString(DBO), name);
```

Comment:
default_schema? instead of dbo everytime.

Reply:
I am wondering if it is better to skip this currently and fix it separately when I fix BABEL-4813 which will also be the long term solution.

Currently the problems are
1. master..p will always resolve to master.dbo.p (never picks default schema) https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/aed2667f0848571dd22cda006bdfefe4a9883ab2/contrib/babelfishpg_tsql/src/tsqlIface.cpp#L8569
2. The search path inside functions and procedure is currently always set as 
schema which contains the routine, dbo <-- this should have been default schema

So it seems keeping this dbo for now will feel more consistent.